proton-j/src/main/java/org/apache/qpid/proton/message/impl/MessageImpl.java

```diff
@@ -49,11 +49,7 @@
       }
     }
 
-    private static final ThreadLocal<EncoderDecoderPair> tlsCodec = new ThreadLocal<EncoderDecoderPair>() {
-          @Override protected EncoderDecoderPair initialValue() {
-            return new EncoderDecoderPair();
-          }
-      };
+    private static final ThreadLocal<EncoderDecoderPair> tlsCodec = ThreadLocal.withInitial(EncoderDecoderPair::new);
```

Comment:
I spend about half of yesterday chasing down some sporadic test failures I was seeing in the python test suite, that seemed like it must be from local changes I had in that area. Instead, it eventually turned out to be this change, as I also had this commit in my tree to rebase it. For whatever reason, altering this seems to cause a behaviour shift that the Jython based tests don't like (at least in my env), leading to sporadic failure to run any of the tests. I'll be committing an updated version without this change.

Reply:
Well, I wouldn't have guessed this is not equivalent change. Another lesson in being conservative and not changing what is known to work without pressing reasons. (Previously I had a bad experience trying to upgrade JUnit version once, with random changes in test results.)